src/decisionengine_modules/graphite_client.py

```diff
@@ -25,8 +25,9 @@ def __init__(self, host="fifemondata.fnal.gov", pickle_port=2004, logger=None):
     def send_dict(self, namespace, data, debug_print=True, send_data=True):
         """send data contained in dictionary as {k: v} to graphite dataset
         $namespace.k with current timestamp"""
-        if data is None and self.logger is not None:
-            self.logger.warning("Warning: send_dict called with no data")
+        if data is None:
+            if self.logger is not None:
```

Comment:
I believe the complexity scanning tools prefer a single `if` with an `and` joining the two.

Reply:
But the logic is subtly different--`if data is None` evaluates to `True`, the method must return regardless of whether `self.logger is not None`.  The diff obscures this nuance.

Reply:
Ahh, I see now.